rbac/src/test/java/org/jboss/hal/testsuite/test/rbac/ServerManagementTestCase.java

```diff
@@ -51,98 +60,106 @@ public static void afterClass() {
+            boolean buttonShouldBeVisible) throws Exception {
+        if (ops.isServerRunning(host, server)) {
+            changeServerState(host, server, ServerState.STOPPED, serverShouldBeVisible, buttonShouldBeVisible);
+            changeServerState(host, server, ServerState.RUNNING, serverShouldBeVisible, buttonShouldBeVisible);
+        } else {
+            changeServerState(host, server, ServerState.RUNNING, serverShouldBeVisible, buttonShouldBeVisible);
+            changeServerState(host, server, ServerState.STOPPED, serverShouldBeVisible, buttonShouldBeVisible);
         }
     }
 
-    public void startServer(boolean shouldSucceed, String host, String server) {
-        boolean missingBtn = false;
-        navigation = new FinderNavigation(browser, DomainRuntimeEntryPoint.class)
+    private void changeServerState(String host, String server, ServerState serverState, boolean serverShouldBeVisible,
+            boolean buttonShouldBeVisible) throws Exception {
+        boolean buttonVisibility = true;
+        boolean serverVisibility = true;
+        FinderNavigation navigation = new FinderNavigation(browser, DomainRuntimeEntryPoint.class, 100)
                 .addAddress(FinderNames.BROWSE_DOMAIN_BY, FinderNames.HOSTS)
                 .addAddress(FinderNames.HOST, host)
-                .addAddress("Server", server);
+                .addAddress(FinderNames.SERVER, server);
 
         try {
-           navigation.selectRow().invoke("Start");
+           navigation.selectRow().invoke(serverState.splitButtonOption);
+           Console.withBrowser(browser).openedWindow(ConfirmationWindow.class).confirm().assertClosed();
+           Console.withBrowser(browser).waitUntilLoaded();
         } catch (NoSuchElementException ex) {
-            missingBtn = true;
+            buttonVisibility = false;
+        } catch (TimeoutException ex) {
+            serverVisibility = false;
         }
 
-         assertEquals("Missing start btn for server - " + server + " in host - " + host, missingBtn, !shouldSucceed);
+        assertEquals("Server '" + server + "' of host '" + host + "' visibility was '" + serverVisibility
+                + "' but should be '" + serverShouldBeVisible + "'.", serverShouldBeVisible, serverVisibility);
 
-        if (shouldSucceed) {
-            address = new ResourceAddress(new ModelNode("/host=" + host + "/server=" + server));
+        if (serverShouldBeVisible) {
+            assertEquals("Missing " + serverState.splitButtonOption + " option for server '" + server + "' of host '"
+                    + host + "'.", buttonShouldBeVisible, buttonVisibility);
 
-            verifier.verifyAttribute(address, "server-state", "running");
+            if (buttonShouldBeVisible) {
+                Address address = ops.getServerAddress(host, server);
+                new ResourceVerifier(address, client, 4000).verifyAttribute("server-state", serverState.attrValue);
```

Comment:
Why 4000 ms? If this is common timeout for multiple cases when using ResourceVerifier, it would make sense putting it to some constant.